### PR TITLE
Integration of Mulitplicity Histos

### PR DESCRIPTION
**Pull request by makor**
Also sign fix in  PWGGA/Hyperon/macros/AddTaskSigma0Run2syst.C.

### CODE REVIEW

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 1]
```diff
@@ -55,11 +55,13 @@ ClassImp(AliSigma0PhotonMotherCuts)
       fHistInvMassEta(nullptr),
       fHistEtaPhi(nullptr),
       fHistPtY(),
+      fHistPtMult(),
       fHistArmenterosBefore(nullptr),
       fHistArmenterosAfter(nullptr),
       fHistMixedPtY(),
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]:
```suggestion
       fHistMixedPtY(),
      fHistMixedPtMult(),
```

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 2]
```diff
@@ -134,11 +136,13 @@ AliSigma0PhotonMotherCuts::AliSigma0PhotonMotherCuts(
...
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]:
```suggestion
      fHistMixedPtY(),
      fHistMixedPtMult(),
```

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 3]
```diff
@@ -285,12 +289,15 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGamma(
...
       if (std::abs(rap) > fRapidityMax) continue;
       const int rapBin = GetRapidityBin(rap);
+      const int multBin = GetMultiplicityBin(lPercentile);
       if (!fIsLightweight) {
         fHistArmenterosAfter->Fill(armAlpha, armQt);
         fHistInvMassRecPhoton->Fill(pT, sigma.GetRecMassPhoton());
         fHistInvMassRecLambda->Fill(pT, sigma.GetRecMassLambda());
         fHistInvMassRec->Fill(pT, sigma.GetRecMass());
         if (rapBin > -1) fHistPtY[rapBin]->Fill(pT, invMass);
+        if (multBin == -1) continue;
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]:
```suggestion
        if (multBin < 0) continue;
```

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 4]
```diff
@@ -364,7 +380,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEvent(
         fHistMixedInvMassPt->Fill(pT, invMass);
         if (!fIsLightweight) {
           const int rapBin = GetRapidityBin(rap);
+          const int multBin = GetMultiplicityBin(lPercentile);
           if (rapBin > -1) fHistMixedPtY[rapBin]->Fill(pT, invMass);
+          if (multBin == -1) continue;
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]:
```suggestion
          if (multBin < 0) continue;
```

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 5]
```diff
@@ -393,7 +412,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEvent(
...
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]:
```suggestion
          if (multBin < 0) continue;
```

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 6]
```diff
@@ -414,11 +436,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEventBinned(
   }
 
   const float zVertex = fInputEvent->GetPrimaryVertex()->GetZ();
-
   const int zVertexBin = GetZvertexBin(zVertex);
   const int multBin = GetMultiplicityBin(lPercentile);
-
-  if (zVertexBin < 0 || multBin < 0) return;
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]: Why the change? Was fine before

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 7]
```diff
@@ -514,7 +537,8 @@ void AliSigma0PhotonMotherCuts::FillEventBuffer(
   const int zVertexBin = GetZvertexBin(zVertex);
   const int multBin = GetMultiplicityBin(lPercentile);
 
-  if (zVertexBin < 0 || multBin < 0) return;
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]: Why the change?

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 8]
```diff
@@ -781,6 +824,31 @@ void AliSigma0PhotonMotherCuts::InitCutHistograms(TString appendix) {
     fHistograms->SetName(appendix);
   }
 
+  for (int i = 0; i < static_cast<int>(multBins.size() - 1); i++) {
+    fHistPtMult[i] =
+        new TH2F(Form("fHistPtMult_%.2f_%.2f", multBins[i], multBins[i + 1]),
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]:
```suggestion
        new TH2F(Form("fHistPtMult_%i", i]),
```
To remove the overhead and make the output less chaotic

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 9]
```diff
@@ -781,6 +824,31 @@ void AliSigma0PhotonMotherCuts::InitCutHistograms(TString appendix) {
...
+                 Form("%.2f < <N> < %.2f ; #it{p}_{T} (GeV/#it{c}); "
+                      "M_{#Lambda#gamma} (GeV/#it{c}^{2})",
+                      multBins[i], multBins[i + 1]),
+                 100, 0, 10, 150, 1.15, 1.3);
+    fHistograms->Add(fHistPtMult[i]);
+    fHistMixedPtMult[i] = new TH2F(
+        Form("fHistMixedPtMult%.2f_%.2f", multBins[i], multBins[i + 1]),
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]:
```suggestion
        Form("fHistMixedPtMult_%i", i),
```

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 10]
```diff
@@ -781,6 +824,31 @@ void AliSigma0PhotonMotherCuts::InitCutHistograms(TString appendix) {
...
+        Form("%.2f < <N> < %.2f ; #it{p}_{T} (GeV/#it{c}); "
+             "M_{#Lambda#gamma} (GeV/#it{c}^{2})",
+             multBins[i], multBins[i + 1]),
+        100, 0, 10, 150, 1.15, 1.3);
+    fHistograms->Add(fHistMixedPtMult[i]);
+    fHistMixedInvMassBinnedMultPt[i] =
+        new TH2F(Form("fHistMixedInvMassBinnedMultPt%.2f_%.2f", multBins[i],
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]:
```suggestion
        new TH2F(Form("fHistMixedInvMassBinnedMultPt_%i", i
```

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.h [hunk 11]
```diff
@@ -129,13 +129,13 @@ class AliSigma0PhotonMotherCuts : public TObject {
   int fPDGDaughter1;   //
   int fPDGDaughter2;   //
 
-  float fMassSigma;         //
-  float fSigmaMassCut;      //
-  float fSidebandCutUp;     //
-  float fSidebandCutDown;   //
-  float fPhotonPtMin;       //
-  float fPhotonPtMax;       //
-  float fRapidityMax;       //
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.h]: not so nice to just commit empty spaces due to the formatting

**Review by a-mathis — CHANGES_REQUESTED** [review]

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.h (hunk 11 above)
Comment (makor) [reply PWGGA/Hyperon/AliSigma0PhotonMotherCuts.h]: I ran the beautifier, should I correct it by hand?

Comment (a-mathis) [reply PWGGA/Hyperon/AliSigma0PhotonMotherCuts.h]: Please

Comment (a-mathis) [reply PWGGA/Hyperon/AliSigma0PhotonMotherCuts.h]: Can you fix this please

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 12]
```diff
@@ -393,7 +414,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEvent(
         fHistMixedInvMassPt->Fill(pT, invMass);
         if (!fIsLightweight) {
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]: The histograms are initialised anyways, so could you fill them all the time (not only when fIsLightweight is true

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 13]
```diff
@@ -364,7 +382,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEvent(
...
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]: The histograms are initialised anyways, so could you fill them all the time (not only when fIsLightweight is true

File: PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx [hunk 14]
```diff
@@ -285,12 +291,15 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGamma(
 
       if (std::abs(rap) > fRapidityMax) continue;
       const int rapBin = GetRapidityBin(rap);
+      const int multBin = GetMultiplicityBin(lPercentile);
       if (!fIsLightweight) {
```
Comment (a-mathis) [inline PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx]: The histograms are initialised anyways, so could you fill them all the time (not only when fIsLightweight is true
Just move them to where fHistInvMassPt is filled (and accordingly for the others)

**Review by a-mathis — CHANGES_REQUESTED** [review]

**Review by a-mathis — APPROVED** [review]
